src/analysis/calculus/specific_functions.lean

```diff
@@ -0,0 +1,193 @@
+  { convert (tendsto_pow_mul_exp_neg_at_top_nhds_0 (2 * n)).comp tendsto_inv_zero_at_top,
+    ext x,
+    field_simp },
+  convert A.mul B;
+  simp [mul_div_assoc]
+end
+
+/-- Deduce from the limiting behavior at `0` of its derivative and general differentiability
+extension theorems that the auxiliary function `exp_neg_inv_glue_aux n` is differentiable at `0`,
+with derivative `0`. -/
+lemma exp_neg_inv_glue_aux_deriv_zero (n : ℕ) :
+  has_deriv_at (exp_neg_inv_glue_aux n) 0 0 :=
+begin
+  -- we check separately differentiability on the left and on the right
+  have A : has_deriv_within_at (exp_neg_inv_glue_aux n) (0 : ℝ) (Iic 0) 0,
+  { apply (has_deriv_at_const (0 : ℝ) (0 : ℝ)).has_deriv_within_at.congr,
+    { assume y hy,
+      simp at hy,
+      simp [exp_neg_inv_glue_aux, hy] },
+    { simp [exp_neg_inv_glue_aux, le_refl] } },
+  have B : has_deriv_within_at (exp_neg_inv_glue_aux n) (0 : ℝ) (Ici 0) 0,
+  { have diff : differentiable_on ℝ (exp_neg_inv_glue_aux n) (Ioi 0) :=
+      λx hx, (exp_neg_inv_glue_aux_deriv_pos n x hx).differentiable_at.differentiable_within_at,
+    -- next line is the nontrivial bit of this proof, appealing to differentiability
+    -- extension results.
+    apply has_deriv_at_interval_left_endpoint_of_tendsto_deriv diff _ self_mem_nhds_within,
+    { refine (exp_neg_inv_aux_limit (n+1)).congr' _,
+      apply mem_sets_of_superset self_mem_nhds_within (λx hx, _),
+      simp [(exp_neg_inv_glue_aux_deriv_pos n x hx).deriv] },
+    { have : exp_neg_inv_glue_aux n 0 = 0, by simp [exp_neg_inv_glue_aux, le_refl],
+      simp only [continuous_within_at, this],
+      refine (exp_neg_inv_aux_limit n).congr' _,
+      apply mem_sets_of_superset self_mem_nhds_within (λx hx, _),
+      have : ¬(x ≤ 0), by simpa using hx,
+      simp [exp_neg_inv_glue_aux, this] } },
+  simpa using A.union B,
+end
+
+/-- At every point, the auxiliary function `exp_neg_inv_glue_aux n` has a derivative which is
+equal to `exp_neg_inv_glue_aux (n+1)`. -/
+lemma exp_neg_inv_glue_aux_has_deriv_at (n : ℕ) (x : ℝ) :
+  has_deriv_at (exp_neg_inv_glue_aux n) (exp_neg_inv_glue_aux (n+1) x) x :=
+begin
+  -- check separately the result for `x < 0`, where it is trivial, for `x > 0`, where it is done
+  -- in `exp_neg_inv_glue_aux_deriv_pos`, and for `x = 0`, done in
+  -- `exp_neg_inv_glue_aux_deriv_zero`.
+  rcases lt_trichotomy x 0 with hx|hx|hx,
+  { have : exp_neg_inv_glue_aux (n+1) x = 0, by simp [exp_neg_inv_glue_aux, le_of_lt hx],
+    rw this,
+    apply (has_deriv_at_const x (0 : ℝ)).congr_of_mem_nhds,
+    have : Iio (0 : ℝ) ∈ 𝓝 x := gt_mem_nhds hx,
+    filter_upwards [this],
+    assume y hy,
+    simp [exp_neg_inv_glue_aux, le_of_lt hy] },
+  { have : exp_neg_inv_glue_aux (n + 1) 0 = 0, by simp [exp_neg_inv_glue_aux, le_refl],
+    rw [hx, this],
+    exact exp_neg_inv_glue_aux_deriv_zero n },
+  { have : exp_neg_inv_glue_aux (n+1) x
+      = (exp_neg_inv_aux_poly (n+1)).eval x * exp (-x⁻¹) / x^(2 * (n+1)),
```

Comment:
```suggestion
  { have : exp_neg_inv_glue_aux (n+1) x =
      (exp_neg_inv_aux_poly (n+1)).eval x * exp (-x⁻¹) / x^(2 * (n+1)),
```
I think the style guide wants the line break after the equals sign.

Reply:
Now it fits on one line.